workers/unity/Packages/com.improbable.gdk.core/GameObjectRepresentation/Injection/WorldCommandRequestSender.cs

```diff
@@ -94,6 +123,71 @@ public long ReserveEntityIds(uint numberOfEntityIds, uint? timeoutMillis = null,
                     return request.RequestId;
                 }
 
+                /// <summary>
+                ///     Sends a CreateEntity command request.
+                /// </summary>
+                /// <param name="entityTemplate">
+                ///     The EntityTemplate object that defines the SpatialOS components on the to-be-created entity.
+                /// </param>
+                /// <param name="callback">
+                ///     A callback that will be called with the command response.
+                /// </param>
+                /// <param name="timeoutMillis">
+                ///     (Optional) The command timeout in milliseconds. If not specified, will default to 5 seconds.
+                /// </param>
+                /// <returns>The request ID of the command request.</returns>
+                public long CreateEntity(EntityTemplate entityTemplate, Action<CreateEntity.ReceivedResponse> callback,
```

Comment:
can we just merge the two methods and use the following method signature:
`public long CreateEntity(EntityTemplate entityTemplate,  Action<CreateEntity.ReceivedResponse> callback,EntityId entityId = null, uint? timeoutMillis = null)`

Reply:
We can but I split them out because the entity ID is configuration, it's part of the payload. We do the odd things of having world command payloads as individual parameters and component commands as a struct.

So yes, but it means you are putting a callback inside the payload parameters.

Reply:
not sure I understand what exactly the difference would be?

also, I'm not sure what to think about putting the callback into the context object...
What if you would want to depend on the context when the callback gets triggered. Or have extra information while still triggering the callback

Reply:
You want to keep parameters grouped based on what they are to make it clear what they are for. If you mix configuration and data it makes the api a lot more confusing. At a glance, if I didn’t know how the commands all work, I’d have thought an optional entity I’d parameter, after the callback, would be the entity ID that wants the response or something. Arguably that means those parameters should be a struct, but I didn’t have time to change things like that as a drive by.

A better callback api would give you access to a context object too, however you can also just capture that in your callback, which is what expects you to do. I also doubt there are many use cases for mixing closures and handlers that need context beyond their request ID.

But ultimately this is a hack being made for a specific purpose, before work is done that makes most of it moot. Which is why this most just needs to exist rather than be perfect.

Reply:
hm... can't say I like that we are now hacking this together instead of properly designing it. It looks like we should rethink the internal implementation instead of adding hacks on top of it.

though I get  why we are doing it and the hack is not too big, so I guess it's fine for now...

Reply:
What is currently preventing us from using a struct for world command payloads instead of parameters to these methods?

Reply:
OK, it will need to be considered in future polish passes it seems.